### PR TITLE
remove go-ens / go-ethereum dependency

### LINKED ISSUE #512: [Platform] Replace go-ethereum with lightweight alternative

### PR DESCRIPTION
**Pull request by hewigovens**
Fix #512 and reduce built binary size (10m+)

### CODE REVIEW

File: platform/ethereum/domain.go [hunk 1]
```diff
@@ -1,11 +1,11 @@
 package ethereum
 
 import (
-	cc "github.com/hewigovens/go-coincodec"
 	CoinType "github.com/trustwallet/blockatlas/coin"
 	"github.com/trustwallet/blockatlas/pkg/blockatlas"
 	"github.com/trustwallet/blockatlas/pkg/errors"
 	"github.com/trustwallet/blockatlas/pkg/logger"
+	cc "github.com/trustwallet/ens-coincodec"
```
Comment (vikmeup) [inline platform/ethereum/domain.go]: any chance to not use `cc` ?

File: platform/ethereum/base.go [hunk 2]
```diff
@@ -10,14 +10,18 @@ type Platform struct {
...
 
 func Init(coin uint, api, rpc string) *Platform {
-	return &Platform{
+	p := &Platform{
 		CoinIndex: coin,
 		RpcURL:    rpc,
 		client:    Client{blockatlas.InitClient(api)},
+		ens:       RpcClient{blockatlas.InitClient(rpc)},
 	}
+	p.ens.Headers["Content-Type"] = "application/json"
```
Comment (vikmeup) [inline platform/ethereum/base.go]: can we add these headers on JSON RPC clients and remove from else where?

**Comment by catenocrypt** [comment]
So trustwallet/ens-go is up to the task?

**Comment by hewigovens** [comment]
> So trustwallet/ens-go is up to the task?

didn't get it, we're good now, no big dependency

**Comment by EnoRage** [comment]
Great improvement

**Comment by EnoRage** [comment]
Sorry, please do not merge it yet, I need to check some stuff

**Comment by EnoRage** [comment]
I decided to run wrk to check to perfomance and stabilty:

``` 
wrk -t2 -c5 -d60s --timeout 2s http://localhost:8420/v2/ns/lookup\?name\=vitalik.eth\&coins\=60 
```

Old version:

```
Running 1m test @ http://localhost:8420/v2/ns/lookup?name=vitalik.eth&coins=60
  2 threads and 5 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   987.50ms  256.85ms   1.94s    85.56%
    Req/Sec     2.26      2.71    10.00     88.83%
  192 requests in 1.00m, 90.80KB read
  Socket errors: connect 0, read 0, write 0, timeout 12
  Non-2xx or 3xx responses: 3
Requests/sec:      3.19
Transfer/sec:      1.51KB

```

New version: 

```
Running 1m test @ http://localhost:8420/v2/ns/lookup?name=vitalik.eth&coins=60
  2 threads and 5 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   429.69ms  208.00ms   1.74s    89.16%
    Req/Sec     5.15      2.76    10.00     62.58%
  526 requests in 1.00m, 246.56KB read
  Socket errors: connect 0, read 0, write 0, timeout 4
Requests/sec:      8.75
Transfer/sec:      4.10KB

```

So, there is more than 2x speedup and less a bit less errors, that is great!

**Comment by catenocrypt** [comment]
> I decided to run wrk to check to perfomance and stabilty:
> 
> ```
> wrk -t2 -c5 -d60s --timeout 2s http://localhost:8420/v2/ns/lookup\?name\=vitalik.eth\&coins\=60 
> ```
> 
> Old version:
> 
> ```
> Running 1m test @ http://localhost:8420/v2/ns/lookup?name=vitalik.eth&coins=60
>   2 threads and 5 connections
>   Thread Stats   Avg      Stdev     Max   +/- Stdev
>     Latency   987.50ms  256.85ms   1.94s    85.56%
>     Req/Sec     2.26      2.71    10.00     88.83%
>   192 requests in 1.00m, 90.80KB read
>   Socket errors: connect 0, read 0, write 0, timeout 12
>   Non-2xx or 3xx responses: 3
> Requests/sec:      3.19
> Transfer/sec:      1.51KB
> ```
> 
> New version:
> 
> ```
> Running 1m test @ http://localhost:8420/v2/ns/lookup?name=vitalik.eth&coins=60
>   2 threads and 5 connections
>   Thread Stats   Avg      Stdev     Max   +/- Stdev
>     Latency   429.69ms  208.00ms   1.74s    89.16%
>     Req/Sec     5.15      2.76    10.00     62.58%
>   526 requests in 1.00m, 246.56KB read
>   Socket errors: connect 0, read 0, write 0, timeout 4
> Requests/sec:      8.75
> Transfer/sec:      4.10KB
> ```
> 
> So, there is more than 2x speedup and a bit less errors, that is great!

Great! +1

**Pull request merged**